Purchases/Public/RCPurchases.h

```diff
@@ -59,6 +61,10 @@ typedef void (^RCPaymentDiscountBlock)(SKPaymentDiscount * _Nullable, NSError *
 NS_SWIFT_NAME(Purchases)
 @interface RCPurchases : NSObject
 
+// TODO during migration make these protected
+@property (nonatomic) RCSubscriberAttributesManager *subscriberAttributesManager;
+@property (nonatomic) RCOperationDispatcher *operationDispatcher;
+
```

Comment:
i tried that but then the test classes get errors (`-[RCPurchases setSubscriberAttributesManager:]: unrecognized selector sent to instance `) when initializing RCPurchases. i think that's because those two were defined in an RCPurchases block within the RCPurchases+SubscriberAttributes.h file?:

`@interface RCPurchases ()

@property (nonatomic) RCSubscriberAttributesManager *subscriberAttributesManager;
@property (nonatomic) RCOperationDispatcher *operationDispatcher;

@end`

Reply:
right... 

So the difference between the two is that `RCPurchases+SubscriberAttributes.h` defined the properties in an extension, whereas in `RCPurchases+Protected.h`, they're defined in a Category. 

The distinction is the lack of a name when doing `@interface RCPurchases ()` vs `@interface RCPurchases (Protected)`. 

Properties defined in the `()` variant will automatically get synthetized, whereas the category variant won't. 

So for `RCPurchases+Protected.h`, this means you'd have to declare it in the header, and then also declare it in the .m (we do that for other properties in that header). 

It doesn't matter much since we'll be getting rid of all of these shortly

Reply:
no action needed, just wanted to provide context

Reply:
I seeeeee, thank you for clarifying!!